src/realm/realmReducer.js

```diff
@@ -28,6 +30,14 @@ const initialState = {
 const convertRealmEmoji = (data): RealmEmojiById =>
   objectFromEntries(Object.keys(data).map(id => [id, { ...data[id], code: id.toString() }]));
 
+const formatJitsiServerUrl = (url: ?string) => {
+  if (url == null) {
+    return null;
+  }
+
+  return url.endsWith('/') ? url : `${url}/`;
```

Comment:
I noticed that the Jitsi server URL has a trailing slash but wasn't sure if I could always rely on that being present, so I did this normalization to avoid the scenario where I create a meeting URL with two slashes between the server URL and the call ID, i.e. `meet.jit.si//video-call-id`.

Reply:
Got it, thanks.

One reason this stood out to me is that there doesn't seem to be any corresponding logic in the webapp. And indeed it looks like the webapp currently has the double-slash behavior -- that's what I get with the "video" button on chat.zulip.org. So given that we're diverging from the webapp's behavior in order to correct a glitch that's seen there, it'd be good to file a quick issue on zulip/zulip so it can be fixed there too.

For handling it in our own code, let's put this logic at the spot where we assemble the full URL. That keeps the reducer simple; and I think the meaning of this slash-fixup becomes a lot easier to clearly understand in that context. Outside of that context, it's not entirely clear what the semantics of this slightly-modified URL are meant to be -- so we end up with two things named `jitsiServerUrl` or `jitsi_server_url` that are very subtly different and it's hard to tell what distinction is intended.

(Further background that I think isn't actionable right now: we've also been wanting to get away from attempting to manipulate URLs as strings, and use `URL` objects instead to do it in a structured way. This kind of string manipulation is exactly how most of our code currently does it, so you've followed the existing style perfectly fine; but see #4146 for discussion of the plan and #4081 for motivation. The webapp has also recently started moving in the same direction. Ultimately we might write something like `const videoCallUrl = new URL(videoCallId.toString(), jitsiServerUrl)` in place of any fiddling with `'/'` or string concatenation here. But that should probably come after discussion on the zulip/zulip issue to be filed about this, because if for some reason the `jitsi_server_url` looks like `https://jitsi.example.com/foo`, with a nonempty path not ending in `/`, then that would have a different behavior from the current code.)

Reply:
That makes sense, I'll file an issue in zulip/zulip repo to track that. I agree that it would be more clear to perform the normalization closer to where the full URL is constructed, I'll move that over to there as well.

As for the discussion on migrating to using WHATWG URL objects for creating and manipulating URLs: I think that's a great idea.

Reply:
Thanks! Looks like that was https://github.com/zulip/zulip/issues/16176 -- adding the link here for cross-reference.